uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/AbsenceType.tpl

```diff
@@ -270,11 +274,58 @@
                       {literal}
                   }
 
+                function initCalculationUnitControls() {
+                  setEntitlementLabelText();
+                  var prev_calculation_unit_val = $('.absence-calculation-unit select option:selected').val();
+                  $('.absence-calculation-unit select').on('change', function(e) {
+                    var add_public_holiday_entitlement = $('.public-holiday-option input[type=radio]:checked').val();
+                    var hours_unit_value = getHoursUnitValue();
+                    if(this.value == hours_unit_value && add_public_holiday_entitlement == 1) {
+                      $('.absence-calculation-unit select').select2('val', prev_calculation_unit_val);
+                      showPublicHolidayEntitlementErrorAlert();
+                    }
+                    else{
+                      prev_calculation_unit_val = $('.absence-calculation-unit select option:selected').val();
+                    }
+                    setEntitlementLabelText();
+                  });
+                }
+
+                function setEntitlementLabelText() {
+                  $('.entitlement-label').text($('.absence-calculation-unit select option:selected').text());
+                }
+
+                function initPublicHolidayControls() {
```

Comment:
Instead of all this complex logic, what about simply disabling the public holiday checkbox when the calculation unit is Hours?

Reply:
There are different scenarios to this, E.g user  selects `Days` as the calculation unit value, then he selects `add public holiday entitlement` as yes. At this point we will not disable the field because its a valid change, Then the user is now able to select The Hours option value, which is not allowed.

Reply:
Sorry, I think I wasn't clear. What I meant was:
- User selects `Days` as the calculation unit
- Next, selects `add public holiday to entitlement`
- Next changes her mind and select `Hours` as the calculation unit
- We automatically disable the `add public holiday to entitlement` checkbox and reset value to its default one
- User then decides to go back to `Days` again
- We enable the field once more

I think this is simpler because it's just a matter of enabling and disabling fields and we won't need to handle the click on the public holiday field. However, the user won't be informed of anything, but perhaps that's ok. Especially because the specification doesn't mention anything about this message. Can you please check with Mike if we should have the message or not?

Reply:
Note that for this to work, we need to make sure to also disable the field right after opening form to edit an Absence Type which the calculation unit is hours